sklearn/preprocessing/tests/test_encoders.py

```diff
@@ -497,6 +497,25 @@ def test_one_hot_encoder_feature_names_unicode():
     assert_array_equal([u'n👍me_c❤t1', u'n👍me_dat2'], feature_names)
 
 
+@pytest.mark.parametrize("X", [np.array([[1, np.nan]]).T,
+                               np.array([['a', np.nan]], dtype=object).T],
+                         ids=['numeric', 'object'])
+@pytest.mark.parametrize("handle_unknown", ['error', 'ignore'])
+def test_one_hot_encoder_raise_missing(X, handle_unknown):
+    ohe = OneHotEncoder(categories='auto', handle_unknown=handle_unknown)
+
+    with pytest.raises(ValueError):
+        ohe.fit(X)
```

Comment:
I though (reading through your issue) this error would be raised only for `X.dtype == 'object'` not e.g. 'float'?

Reply:
for float it already raises correctly on master (on a first check_array when calling `fit`), it was only not explicitly tested. So the bug was only for object dtype.